frontend/src/components/auth/Signup/index.tsx

```diff
@@ -51,6 +64,17 @@ const Signup = () => {
           navigation={navigation}
         />
       );
+    case "terms conditions":
+      if (formValues.role === Role.VOLUNTEER) {
+        return (
+          <TermsConditions
+            formData={formValues}
+            setForm={setForm}
+            navigation={navigation}
+          />
+        );
+      }
+      return next();
```

Comment:
we shouldn't have the steps be conditional here, rather you should skip it inside of the donor account details, so change it back to 
```suggestion
        return (
          <TermsConditions
            formData={formValues}
            setForm={setForm}
            navigation={navigation}
          />
   )
```

and then in https://github.com/uwblueprint/community-fridge-kw/blob/88498b31174adb59890e9cf2d3e4f9bc5475e160/frontend/src/components/auth/Signup/AccountDetails.tsx#L106
you can do: 
```typescript
    return role===Role.VOLUNTEER ? go("terms conditions") : go("email verification");
```